### DISCUSSION TITLE
Possible to adjust Anime Matrix picture in future updates?

### DISCUSSION THREAD

**Discussion by chorong761** [seerge/g-helper#107] [Ideas]
Would this be a possible new feature in a future update to adjust the image (e.g. zoom, rotation, etc) added to the Anime Matrix display?

**Comment by seerge** [comment]
I leave this adjustments to the end user :) you can edit your picture / gif as you want. 
Current app logic - it fits / resizes whole picture to the matrix dimensions and aligns it to the right of the screen (where it's more pixels)

**Reply by chorong761** [reply]
I see. Are there any known conflicts with Ghelper and AC when it comes to anime matrix? I wanted to try Ghelper before removing AC, but adding e.g. [this](https://www.daebak.de/media/image/cf/e8/b3/csr_logo1.png) image into Ghelper and I can't even tell it is the same image when looking at the Anime Matrix display. I've tried some other images (transparent and non-transparent) and it gave me similar results.

**Reply by seerge** [reply]
There is no "conflicts", just both apps can set animatrix :) As for your picture - you need to "prepare" it and make black background / white color before applying. What you linked will look inverted ofc. Also it's too much fine details imho. Animatrix resolution is very low, so try to use something quite basic :)

**Reply by chorong761** [reply]
Thanks, I will give it a try. I wasn't expecting I needed to do that since it worked well in AC.

**Reply by chorong761** [reply]
> There is no "conflicts", just both apps can set animatrix :) As for your picture - you need to "prepare" it and make black background / white color before applying. What you linked will look inverted ofc. Also it's too much fine details imho. Animatrix resolution is very low, so try to use something quite basic :)

I just tried something that has white text and black background and it is the same.

**Reply by seerge** [reply]
What model of the laptop do you have ?

**Reply by seerge** [reply]
I have made some improvements to the animatrix rendering (so now your text from https://www.daebak.de/media/image/cf/e8/b3/csr_logo1.png is a bit more readable) 

You can try v0.34 https://github.com/seerge/g-helper/releases/download/v0.34/GHelper.zip

**Reply by chorong761** [reply]
Sorry for the late reply. I'm using the 2021 model. 

Here is how it looks like when I apply the image using v0.34 https://i.imgur.com/9it7QvM.png

**Reply by seerge** [reply]
Ok, that explains a lot. Your matrix size is different from mine (on 2022), and i think it shifts all the pixels. Can you try to set a very simple picture, like this one 
![icons8-happy-100](https://user-images.githubusercontent.com/5920850/226494348-61ec0425-d7e0-42e2-866b-5979e56c8f0a.png)
and send me another photo? Thanks!

**Reply by seerge** [reply]
![IMG_4177](https://user-images.githubusercontent.com/5920850/226494716-972c11a5-f313-4e31-8639-dac9723aaf70.jpg)
this is how your picture looks on mine btw

**Comment by chorong761** [comment]
![圖片](https://user-images.githubusercontent.com/17227895/226497457-530909de-54e6-42de-85c0-ba625a0368a8.png)

This is with the smiley face you provided

**Comment by seerge** [comment]
Ok, can you try this build ? :)
[GHelper.zip](https://github.com/seerge/g-helper/files/11024226/GHelper.zip)

**Reply by chorong761** [reply]
![圖片](https://user-images.githubusercontent.com/17227895/226528171-06fcce72-ef2a-4c75-8732-b3b82cb08074.png)

New build

**Reply by seerge** [reply]
Looks like a step in a right direction, can you try this one then ?
[GHelper.zip](https://github.com/seerge/g-helper/files/11026818/GHelper.zip)

**Reply by chorong761** [reply]
![圖片](https://user-images.githubusercontent.com/17227895/226612208-5ff3459f-cd1c-416e-a575-cc7a98b3a9de.png)

**Reply by seerge** [reply]
[GHelper.zip](https://github.com/seerge/g-helper/releases/download/v0.35/GHelper.zip)
@chorong761  Can you try again? I wish I had a device, then it would be a question of 5 minutes to sort out :))

**Reply by chorong761** [reply]
No problem, I'll try it tonight and send you a pic

**Reply by chorong761** [reply]
> [GHelper.zip](https://github.com/seerge/g-helper/releases/download/v0.35/GHelper.zip) @chorong761 Can you try again? I wish I had a device, then it would be a question of 5 minutes to sort out :))

![圖片](https://user-images.githubusercontent.com/17227895/226807742-05f2c7a6-d256-41c0-b144-19ff5789e1cb.png)

**Reply by seerge** [reply]
Ok, and this one 
[GHelper.zip](https://github.com/seerge/g-helper/files/11039353/GHelper.zip)
?

**Reply by chorong761** [reply]
![圖片](https://user-images.githubusercontent.com/17227895/226928578-af5b60eb-cb27-4d7d-8b84-2728c3b601c2.png)

**Reply by seerge** [reply]
[GHelper.zip](https://github.com/seerge/g-helper/files/11040869/GHelper.zip)

and now ? do you have discord or any other messenger btw ? i think it would be way faster to do it there

**Reply by chorong761** [reply]
![圖片](https://user-images.githubusercontent.com/17227895/227224781-6f960cd7-e249-424a-9843-aa1c004de066.png)


Yes, I do have discord. What is your ID?

**Reply by seerge** [reply]
It's Serge#2609 
and try
[GHelper.zip](https://github.com/seerge/g-helper/files/11051755/GHelper.zip)
